ui/components/app/signature-request/signature-request.js

```diff
@@ -0,0 +1,406 @@
+          fromCurrency: nativeCurrency,
+          toCurrency: currentCurrency,
+          conversionRate,
+          numberOfDecimals: 6,
+          toDenomination: EtherDenomination.ETH,
+        }),
+        currentCurrency,
+      )
+    : new Numeric(balance, 16, EtherDenomination.WEI)
+        .toDenomination(EtherDenomination.ETH)
+        .round(6)
+        .toBase(10)
+        .toString();
+
+  const onSign = (event) => {
+    sign(event);
+    trackEvent({
+      category: MetaMetricsEventCategory.Transactions,
+      event: 'Confirm',
+      properties: {
+        action: 'Sign Request',
+        legacy_event: true,
+        type,
+        version,
+      },
+    });
+  };
+
+  const onCancel = (event) => {
+    cancel(event);
+    trackEvent({
+      category: MetaMetricsEventCategory.Transactions,
+      event: 'Cancel',
+      properties: {
+        action: 'Sign Request',
+        legacy_event: true,
+        type,
+        version,
+      },
+    });
+  };
+
+  const messageIsScrollable =
+    messageRootRef?.scrollHeight > messageRootRef?.clientHeight;
+
+  const targetSubjectMetadata = txData.msgParams.origin
+    ? subjectMetadata?.[txData.msgParams.origin]
+    : null;
+
+  return (
+    <div className="signature-request">
+      <ConfirmPageContainerNavigation />
+      <div
+        className="request-signature__account"
+        data-testid="request-signature-account"
+      >
+        <NetworkAccountBalanceHeader
+          networkName={currentNetwork}
+          accountName={name}
+          accountBalance={balanceInBaseAsset}
+          tokenName={
+            conversionRate ? currentCurrency?.toUpperCase() : nativeCurrency
+          }
+          accountAddress={address}
+        />
+      </div>
+      <div className="signature-request-content">
+        {(txData?.securityProviderResponse?.flagAsDangerous !== undefined &&
+          txData?.securityProviderResponse?.flagAsDangerous !==
+            SECURITY_PROVIDER_MESSAGE_SEVERITIES.NOT_MALICIOUS) ||
+        (txData?.securityProviderResponse &&
+          Object.keys(txData.securityProviderResponse).length === 0) ? (
+          <SecurityProviderBannerMessage
+            securityProviderResponse={txData.securityProviderResponse}
+          />
+        ) : null}
+        {
+          ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+          selectedAccount.address === address ? null : (
+            <Box
+              className="request-signature__mismatch-info"
+              display={DISPLAY.FLEX}
+              width={BLOCK_SIZES.FULL}
+              padding={4}
+              marginBottom={4}
+              backgroundColor={BackgroundColor.primaryMuted}
+            >
+              <Icon
+                name={IconName.Info}
+                color={IconColor.infoDefault}
+                marginRight={2}
+              />
+              <Text
+                variant={TextVariant.bodyXs}
+                color={TextColor.textDefault}
+                as="h7"
```

Comment:
No html h7 element headings go from h1-h6
```suggestion
                as="h6"
```